tools/src/test/java/org/apache/kafka/tools/reassign/ReassignPartitionsIntegrationTest.java

```diff
@@ -0,0 +1,897 @@
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.utils.Time;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+import scala.None$;
+import scala.Option;
+import scala.Tuple2;
+import scala.collection.JavaConverters;
+import scala.collection.Seq;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static kafka.test.annotation.Type.KRAFT;
+import static kafka.test.annotation.Type.ZK;
+import static org.apache.kafka.clients.CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.server.common.MetadataVersion.IBP_2_7_IV1;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SuppressWarnings({"ClassDataAbstractionCoupling", "ClassFanOutComplexity"})
+@ExtendWith(value = ClusterTestExtensions.class)
+@Tag("integration")
+@Timeout(300)
+public class ReassignPartitionsIntegrationTest {
+    private static final Map<Integer, String> BROKERS = new HashMap<>();
+
+    private static final Map<String, List<List<Integer>>> TOPICS = new HashMap<>();
+
+    private static final Map<Integer, Map<String, Long>> UNTHROTTLED_BROKER_CONFIGS;
+
+    static {
+        BROKERS.put(0, "rack0");
+        BROKERS.put(1, "rack0");
+        BROKERS.put(2, "rack1");
+        BROKERS.put(3, "rack1");
+        BROKERS.put(4, "rack1");
+
+        TOPICS.put("foo", Arrays.asList(Arrays.asList(0, 1, 2), Arrays.asList(1, 2, 3)));
+        TOPICS.put("bar", Arrays.asList(Arrays.asList(3, 2, 1)));
+        TOPICS.put("baz", Arrays.asList(Arrays.asList(1, 0, 2), Arrays.asList(2, 0, 1), Arrays.asList(0, 2, 1)));
+
+        Map<String, Long> brokerConfig = new HashMap<>();
+
+        ReassignPartitionsCommand.brokerLevelThrottles().foreach(throttle -> {
+            brokerConfig.put(throttle, -1L);
+            return null;
+        });
+
+        UNTHROTTLED_BROKER_CONFIGS = IntStream.range(0, 5).boxed().collect(Collectors.toMap(Function.identity(), brokerId -> brokerConfig));
+    }
+
+    private ClusterInstance cluster;
+
+    private Admin adminClient;
+
+    @ClusterTemplate("zkAndKRaft")
```

Comment:
Can we keep with the @ParameterizedTest format we use for other tests? I think it is good to keep things consistent.

Reply:
Also let me know if this was decided for the Java migrations and I missed that. 😅

Reply:
Hello, @jolshan 

For now, I know the following ways to setup integration test 

* Use scala `QuorumTestHarness` - see existing `ReassignPartitionsIntegrationTest.scala`. 
* Use java annotation `@ClusterTestDefaults` - see `DeleteRecordsCommandTest`.
* Use `@ClusterTemplate` - like I did in `ReassignPartitionsIntegrationTest.java` for now. 

I don't know what is the correct way to implement test.
AFAICU you suggesting to use `QuorumTestHarness` with ParametrizedTest annotation. 
Is this correct?

I had such version of `ReassignPartitionsIntegrationTest.java` but replace it with the current version. So it's will be easy to revert to earlier version :)

Reply:
I was suggesting the QuorumTestHarness approach, but I wasn't sure if there was a unified/preferred way to do the conversions. Unless there are any issues with the QuorumTestHarness approach or there are benefits with the ClusterTemplate one, I think it makes sense to stick with it.

Reply:
I reverted new `ReassignPartitionsIntegrationTest` to use `QuorumTestHarness` so now it's structure is very similary to the scala version. Please, take a look.